xrpl/model/transactions/validations_commons.go

```diff
@@ -81,14 +81,15 @@ func ValidateBaseTransaction(tx FlatTransaction) error {
 }
 
 func ValidateRequiredField(tx FlatTransaction, field string, checkValidity func(interface{}) bool) error {
```

Comment:
I think we should remove the `checkValidity` method since it only adds typechecking and it doesn't validate that the current value is valid. For example, if I run:

```go
tx["Destination"] = ""
ValidateRequiredField(tx, "Destination", typecheck.IsString())
```

It will return okey since the Destination value is string but it is empty, which is invalid on-chain